pkg/agent/agent.go

```diff
@@ -68,23 +74,65 @@ func ListReleases(actionConfig *action.Configuration, namespace string, listLimi
 	return appOverviews, nil
 }
 
-func NewActionConfig(storageForDriver StorageForDriver, token, namespace string) (*action.Configuration, error) {
-	actionConfig := new(action.Configuration)
-	config, err := rest.InClusterConfig()
+func CreateRelease(config Config, name, namespace, valueString string, ch *chart.Chart) (*release.Release, error) {
+	cmd := action.NewInstall(config.ActionConfig)
+	cmd.ReleaseName = name
+	cmd.Namespace = namespace
+	values, err := getValues([]byte(valueString))
 	if err != nil {
 		return nil, err
 	}
-	config.BearerToken = token
-	config.BearerTokenFile = ""
-	clientset, err := kubernetes.NewForConfig(config)
+	release, err := cmd.Run(ch, values)
+	if err != nil {
+		return nil, err
+	}
+	return release, nil
+}
+
+func NewActionConfig(storageForDriver StorageForDriver, config *rest.Config, clientset *kubernetes.Clientset, namespace string) (*action.Configuration, error) {
+	actionConfig := new(action.Configuration)
 	store := storageForDriver(namespace, clientset)
-	actionConfig.RESTClientGetter = nil     // TODO replace nil with meaningful value
-	actionConfig.KubeClient = kube.New(nil) // TODO replace nil with meaningful value
+	restClientGetter := NewConfigFlagsFromCluster(namespace, config)
+	actionConfig.RESTClientGetter = restClientGetter
+	actionConfig.KubeClient = kube.New(restClientGetter)
 	actionConfig.Releases = store
 	actionConfig.Log = klog.Infof
 	return actionConfig, nil
 }
 
+// NewConfigFlagsFromCluster returns ConfigFlags with default values set from within cluster
+func NewConfigFlagsFromCluster(namespace string, clusterConfig *rest.Config) *genericclioptions.ConfigFlags {
+	impersonateGroup := []string{}
+	insecure := false
+
+	// CertFile and KeyFile must be nil for the BearerToken to be used for authentication and authorization instead of the pod's service account.
```

Comment:
Thanks, my first task today was going to be looking for the cause of the service account still being used in a similar PR I had. I'll try the equivalent change but using the `clientcmd` module, which is I think an easier/safer way forward.

Reply:
Great! Let's know how it goes.